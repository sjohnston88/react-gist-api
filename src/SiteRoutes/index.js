import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect } from "react-router-dom";
import App from '../App/index';

class SiteRoutes extends Component {

  // constructor(props) {
  //   super(props);

  //   this.username = this.username.bind(this);
  // }

  render() {
    return (
      <Router>
       <Switch>
          <Route path="/:username" component={App} />
          <Redirect from='/' to='/defunkt' />
        </Switch>
      </Router>
    );
  }
}

export default SiteRoutes;
