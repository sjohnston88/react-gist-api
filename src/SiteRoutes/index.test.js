import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SiteRoutes from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Component/SiteRoutes', () => {
  const setup = () => shallow(<SiteRoutes />);

  it('renders correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot('default');
  });
  
});
