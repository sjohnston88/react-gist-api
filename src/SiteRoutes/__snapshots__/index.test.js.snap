// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SiteRoutes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
                component={[Function]}
                path="/:username"
        />
        <Redirect
                from="/"
                push={false}
                to="/defunkt"
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/:username"
/>,
          <Redirect
            from="/"
            push={false}
            to="/defunkt"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/:username",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "from": "/",
            "push": false,
            "to": "/defunkt",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
                    component={[Function]}
                    path="/:username"
          />
          <Redirect
                    from="/"
                    push={false}
                    to="/defunkt"
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/:username"
/>,
            <Redirect
              from="/"
              push={false}
              to="/defunkt"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/:username",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "from": "/",
              "push": false,
              "to": "/defunkt",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
